src/numeric/mod.rs

```diff
@@ -1 +1,159 @@
+use crate::imp_prelude::*;
+use num_complex::Complex;
+use rawpointer::PointerExt;
+use std::mem;
+use std::ptr::NonNull;
+
 mod impl_numeric;
+
+impl<T, S, D> ArrayBase<S, D>
+where
+    S: Data<Elem = Complex<T>>,
+    D: Dimension,
+{
+    /// Returns views of the real and imaginary components of the elements.
+    ///
+    /// ```
+    /// use ndarray::prelude::*;
+    /// use num_complex::{Complex, Complex64};
+    ///
+    /// let arr = array![
+    ///     [Complex64::new(1., 2.), Complex64::new(3., 4.)],
+    ///     [Complex64::new(5., 6.), Complex64::new(7., 8.)],
+    ///     [Complex64::new(9., 10.), Complex64::new(11., 12.)],
+    /// ];
+    /// let Complex { re, im } = arr.view_re_im();
+    /// assert_eq!(re, array![[1., 3.], [5., 7.], [9., 11.]]);
+    /// assert_eq!(im, array![[2., 4.], [6., 8.], [10., 12.]]);
+    /// ```
+    pub fn view_re_im(&self) -> Complex<ArrayView<'_, T, D>> {
+        debug_assert!(self.pointer_is_inbounds());
+
+        let dim = self.dim.clone();
+
+        // Double the strides. In the zero-sized element case and for axes of
+        // length <= 1, we leave the strides as-is to avoid possible overflow.
+        let mut strides = self.strides.clone();
+        if mem::size_of::<T>() != 0 {
+            for ax in 0..strides.ndim() {
+                if dim[ax] > 1 {
+                    strides[ax] *= 2;
+                }
+            }
+        }
+
+        let ptr_re: NonNull<T> = self.ptr.cast();
+        let ptr_im: NonNull<T> = if self.is_empty() {
+            // In the empty case, we can just reuse the existing pointer since
+            // it won't be dereferenced anyway. It is not safe to offset by one
+            // since the allocation may be empty.
+            self.ptr.cast()
+        } else {
+            // Safe because `self` is nonempty, so we can safely offset into
+            // the first element.
+            unsafe { self.ptr.cast().add(1) }
```

Comment:
```suggestion
            unsafe { ptr_re.add(1) }
```

I would use this for clarity

Reply:
Yes, that's better.